### PR TITLE
Clean up dependencies

### PR DESCRIPTION
**Pull request by eoger**
This PR removes a bunch of useless cargo dependencies whose job could be done by `ring` (openssl, hmac, hkdf, sha256 digest).
Some openssl functions could not be replaced, but thankfully these are only used by the "browserid" code, which is now behind a `browserid` feature flag.

All of this means that we can build fxa_client for iOS without `openssl`, which makes our building process way easier and faster.  
It also drops the size of the iOS universal lib from 40.5mb to 32.4mb, a 20% improvement.

### CODE REVIEW

**Comment by eoger** [comment]
I'm not really sure who to tag for review here, but feel free to drop in and have a look at the changes

File: fxa-rust-client/Cargo.toml [hunk 1]
```diff
@@ -8,20 +8,19 @@ base64 = "0.9.0"
 byteorder = "1.2.3"
 failure = "0.1.1"
 failure_derive = "0.1.1"
-hawk = { git = "https://github.com/eoger/rust-hawk", branch = "use-openssl" }
+hawk = { version = "1.0.2", optional = true }
```
Comment (vladikoff) [inline fxa-rust-client/Cargo.toml]: 👍

File: fxa-rust-client/Cargo.toml [hunk 2]
```diff
@@ -8,20 +8,19 @@ base64 = "0.9.0"
...
 hex = "0.3.1"
-hkdf = "0.4"
-hmac = "0.6.2"
 lazy_static = "1.0.0"
 log = "0.4"
-openssl = "0.10.7"
-rand = "0.5.0-pre.1"
+openssl = { version = "0.10.7", optional = true }
```
Comment (vladikoff) [inline fxa-rust-client/Cargo.toml]: How does that work with Android?

Comment (vladikoff) [reply fxa-rust-client/Cargo.toml]: Oh so it will load it when `browserid` is enabled

**Pull request merged**